### PR TITLE
fn and rpc added

### PR DESCRIPTION
**Pull request by stanieltron**
calculate burn amount fn and rpc call added

### CODE REVIEW

File: pallets/xyk/rpc/src/lib.rs [hunk 1]
```diff
@@ -10,11 +10,11 @@ use sp_runtime::{
     traits::{Block as BlockT, MaybeDisplay, MaybeFromStr},
 };
 use std::sync::Arc;
-use xyk_runtime_api::RpcResult;
 pub use xyk_runtime_api::XykApi as XykRuntimeApi;
+use xyk_runtime_api::{RpcAmountsResult, RpcResult};
 
 #[rpc]
-pub trait XykApi<BlockHash, Balance, ResponseType> {
+pub trait XykApi<BlockHash, Balance, AssetId, ResponseType, ResponseTypeTupple> {
```
Comment (gleb-urvanov) [inline pallets/xyk/rpc/src/lib.rs:17]: Typo: `Tupple` -> `Tuple`
Also, it would be nice to have descriptive names for response types, like "PriceResponse" or "BurnAmountResponse"

File: pallets/xyk/rpc/src/lib.rs [hunk 2]
```diff
@@ -99,4 +110,25 @@ where
...
+
+        let runtime_api_result =
+            api.get_burn_amount(&at, first_asset_id, second_asset_id, liquidity_asset_amount);
+        runtime_api_result.map_err(|e| RpcError {
+            code: ErrorCode::ServerError(1),
+            message: "Unable to serve the request".into(),
+            data: Some(format!("{:?}", e).into()),
+        })
+    }
+}
```
Comment (gleb-urvanov) [inline pallets/xyk/rpc/src/lib.rs:134]: here and later: add new line at the end of the file

File: pallets/xyk/src/lib.rs [hunk 3]
```diff
@@ -535,6 +517,39 @@ impl<T: Trait> Module<T> {
...
+        let second_asset_amount_u256 =
+            second_asset_reserve_u256 * liquidity_asset_amount_u256 / total_liquidity_assets_u256;
+        let second_asset_amount = second_asset_amount_u256
+            .saturated_into::<u128>()
+            .saturated_into::<T::Balance>();
+        let first_asset_amount = first_asset_amount_u256
+            .saturated_into::<u128>()
+            .saturated_into::<T::Balance>();
+
+        return (first_asset_amount, second_asset_amount);
```
Comment (gleb-urvanov) [inline pallets/xyk/src/lib.rs:550]: looks good, can it overflow/underflow somehow? Any other corner cases, e.g. with 1 of 0 amounts?

File: pallets/xyk/src/tests.rs [hunk 4]
```diff
@@ -60,182 +60,18 @@ pub trait Trait: assets::Trait {
 
 fn initialize() {
     // creating asset with assetId 0 and minting to accountId 2
-    let acc_id: u64 = 2;
+    let accId: u64 = 2;
```
Comment (gleb-urvanov) [inline pallets/xyk/src/tests.rs:63]: here and where the `accId` is used: return the variable name to the snake case

File: pallets/xyk/src/tests.rs [hunk 5]
```diff
@@ -60,182 +60,18 @@ pub trait Trait: assets::Trait {
...
+    <pallet_assets::Module<Test>>::assets_issue(&accId, &amount);
+    <pallet_assets::Module<Test>>::assets_issue(&accId, &amount);
     XykStorage::create_pool(
         Origin::signed(2),
         0,
         40000000000000000000,
         1,
         60000000000000000000,
-    )
-    .unwrap();
```
Comment (gleb-urvanov) [inline pallets/xyk/src/tests.rs:75]: Why `unwrap()` is removed? Please return

File: pallets/xyk/src/tests.rs [hunk 6]
```diff
@@ -60,182 +60,18 @@ pub trait Trait: assets::Trait {
...
-}
-
-#[test]
-fn multi() {
```
Comment (gleb-urvanov) [inline pallets/xyk/src/tests.rs:79]: why the test is removed instead of being fixed?

File: pallets/xyk/src/tests.rs [hunk 7]
```diff
@@ -300,10 +136,10 @@ fn create_pool_N_already_exists_other_way() {
 #[test]
 fn create_pool_N_not_enough_first_asset() {
     new_test_ext().execute_with(|| {
-        let acc_id: u64 = 2;
+        let accId: u64 = 2;
```
Comment (gleb-urvanov) [inline pallets/xyk/src/tests.rs:139]: looks like you are using old version of the `tests.rs`
please update your branch

**Review by gleb-urvanov — CHANGES_REQUESTED** [review]

File: pallets/xyk/src/lib.rs [hunk 8]
```diff
@@ -476,10 +459,9 @@ decl_module! {
                 <Pools<T>>::remove((second_asset_id, first_asset_id));
             }
 
-            <assets::Module<T>>::assets_burn(&liquidity_asset_id, &sender, &liquidity_asset_amount)?;
+            <assets::Module<T>>::assets_burn(&liquidity_asset_id, &sender, &liquidity_asset_amount);
```
Comment (gleb-urvanov) [inline pallets/xyk/src/lib.rs:462]: Please resolve warning

**Review by gleb-urvanov — CHANGES_REQUESTED** [review]

**Review by gleb-urvanov — APPROVED** [review]
